### PR TITLE
Vue3 - Adding Sharp Tests

### PR DESCRIPTION
**Pull request by jasonlundien**
This PR will add tests for:  
- sharp icons
- various syntax usage scenarios
- icon aliases

There was also some reordering of the tests to match the Vue 2 testing so the comparison of what tests each repo has would be easier.

Reviewer:
@robmadole --- can you review for accuracy

Dependencies:

release of sharp icons

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
   "contributors": [
     "Travis Chase <travis@fontawesome.com>",
     "Rob Madole <rob@fontawesome.com>",
+    "Jason Lundien <jason@fontawesome.com>",
```
Comment (robmadole) [inline package.json:31]: :heart:

File: src/components/__tests__/FontAwesomeIcon.test.js [hunk 2]
```diff
@@ -3,18 +3,50 @@
...
-  library.add(faCoffee, faCircle, faAlien)
+  library.add(faCoffee, faCircle, faAlien, faGlasses)
 })
 
 afterEach(() => {
   library.reset()
 })
 
+describe('using a family', () => {
+  if(coreHasFeature(REFERENCE_ICON_USING_FAMILY)) {
```
Comment (robmadole) [inline src/components/__tests__/FontAwesomeIcon.test.js]: Just a nit on spacing.

```suggestion
  if (coreHasFeature(REFERENCE_ICON_USING_FAMILY)) {
```

File: src/components/__tests__/FontAwesomeIcon.test.js [hunk 3]
```diff
@@ -3,18 +3,50 @@
...
+    test('will find a sharp-solid-svg-icon with array format', () => {
+      const wrapper = mountFromProps({ icon: ['fass', 'glasses'] })
+
+      expect(wrapper.element.tagName).toBe('svg')
+      expect(wrapper.element.classList.contains('fa-glasses')).toBeTruthy()
+    })
+
+    test('will find a sharp solid icon using short prefix with string format', () => {
+      const wrapper = mountFromProps({ icon: ['fass', 'glasses'] })
```
Comment (robmadole) [inline src/components/__tests__/FontAwesomeIcon.test.js]: Is this test the same as the one above?

File: src/components/__tests__/FontAwesomeIcon.test.js [hunk 4]
```diff
@@ -3,18 +3,50 @@
...
+
+    test('will find a sharp solid icon using long prefix with string format', () => {
+      const wrapper = mountFromProps({ icon: ['fass', 'glasses'] })
+
+      expect(wrapper.element.tagName).toBe('svg')
+      expect(wrapper.element.classList.contains('fa-glasses')).toBeTruthy()
+    })
+
+    test('will find a sharp solid icon using long prefix and style with string format', () => {
+      const wrapper = mountFromProps({ icon: ['fass', 'glasses'] })
```
Comment (robmadole) [inline src/components/__tests__/FontAwesomeIcon.test.js]: Hmm. I think all these tests are the same aren't they?

**Review by robmadole — CHANGES_REQUESTED** [review]

File: src/components/__tests__/FontAwesomeIcon.test.js (hunk 3 above)
Comment (jasonlundien) [reply src/components/__tests__/FontAwesomeIcon.test.js]: yes, I had a couple of duplicate tests.  I went back through and now have things ironed out I believe.

File: src/components/__tests__/FontAwesomeIcon.test.js (hunk 4 above)
Comment (jasonlundien) [reply src/components/__tests__/FontAwesomeIcon.test.js]: yes, I had a couple of duplicate tests.  I went back through and now have things ironed out I believe.

**Pull request closed without merging**